.github/workflows/cs.yml

```diff
@@ -0,0 +1,75 @@
+name: CS
+
+on:
+  # Run on all pushes (except to master) and on all pull requests.
+  push:
+    branches-ignore:
+      - 'master'
+  pull_request:
+  # Allow manually triggering the workflow.
+  workflow_dispatch:
+
+# Cancels all previous workflow runs for the same branch that have not yet completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  checkcs:
+    name: 'Check code style'
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Determine the base branch for the file diff
+        id: base_branch
+        env:
+          BASE_REF: ${{ github.base_ref }}
+        run: |
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            echo "::set-output name=NAME::$BASE_REF"
+            echo "::set-output name=REF::origin/$BASE_REF"
+          else
+            echo '::set-output name=NAME::trunk'
+            echo "::set-output name=REF::origin/trunk"
+          fi
+
+      - name: Fetch base branch
+        run: git fetch --no-tags --depth=1 origin +${{ steps.base_branch.outputs.NAME }}
```

Comment:
TIL: the `+` prefix forces a fetch 😄 . However, I'd prefer to be a bit more explicit about it.
```suggestion
        run: git fetch --force --no-tags --depth=1 origin ${{ steps.base_branch.outputs.NAME }}
```
Also, I'm not sure if it is really needed or if we can do a regular fetch here.
```suggestion
        run: git fetch --no-tags --depth=1 origin ${{ steps.base_branch.outputs.NAME }}
```

Reply:
Tested without the force fetch and seems to work fine, so let's just remove it.